application-workflow-detailed.md

```diff
@@ -0,0 +1,442 @@
+}
+```
+From now on we will refer to this as `RP_PRIVATE_KEY`. The public key looks like:
+
+```json
+{  
+  "jwk":{  
+      "alg":"RS256",
+      "e":"AQAB",
+      "ext":true,
+      "key_ops":[  
+        "verify"
+      ],
+      "kty":"RSA",
+      "n":"mVnn-HwEQi5mZR3Z0Wc7TBrJouOb7acKiseVSkjrj4mWCSEw21VTGNfovzUS71WYKoxFAd8zfkI9-lsAn3tkL8ppcMuI3F8KxsO86nNHSKrxZIlk-bP7RDFfpI9KWyifulKdipEmRit4iN-EFI2mK9KREscPWG083vqn4D81Xe4s0-gmRsBrVanwwu-mTwEKy8RFomV8CXOTcTNntdR3krluXZ38_uKBB1qg6_phBQwZ_sDMXWs8E90eCXhd_EQ6S8PGzCDPT2vg9wCB57ifAXt_8e4ZnqySmFPxegy7j3GcMuyhHzdpvv2fX5DvOxsgkjhsBzby9LD0bxStdBCSFQ"
+  }
+}
+```
+From now on we will refer to this as `RP_PUBLIC_KEY`.
+
+#### 6. Saves the Public/Private key to local storage
+
+The public/private key pair must be saved so that it can be referenced once the OP redirects back to the RP. It is recommended to save it to local storage, but that is not required.
+
+Currently in local storage:
+```
+OPENID_CONFIGURATION
+RP_PRIVATE_KEY
+RP_PUBLIC_KEY
+```
+
+#### 7. Requests OP JWKs
+
+Now that the RP's Public/Private keys are generated. The RP needs to be aware of the OP's public key. To do so a request should be made to the `jwks_uri` from the openid-configuration:
+
+```
+GET https://secureauth.example/jwks
+```
+
+#### 8. Returns OP JWKs
+
+The [JSON Web Key Set (JWKS)](https://auth0.com/docs/jwks) is returned. These will eventually be used to validate the signature of the token the OP will eventually issue. It is recommended that the result of this request should be saved to local storage.
+
+Response body:
+```json
+{  
+  "keys":[  
+    {  
+      "kty":"RSA",
+      "kid":"xeOjes9u3AcU4LBzcanEM7pZLwSlxaN7U62ZzOBDQuw",
+      "alg":"RS256",
+      "key_ops":[  
+        "verify"
+      ],
+      "e":"AQAB",
+      "n":"oB2LgkiZZ5iLAz1d4ua7sVxdbzY2nIRkDtf4UE08mWsD6UYRzLR98_gMAfnKB8i9yPCQkxfA5w_SZq6Y7odG1qSwLHM2mb_O2GSvY9kaG00UpeeEJCR19c7Jkcmq3GXh4yujnm2TFQ6YAzYNgrXkHlusaFUApJaQN6zr4AvmR_vX_5i__Ku7nuU-GbaV75LSr8o0QANdYFF0ooz5DJvydPplF8mO9_oD7ceSNLWP1AXlFs5JH6MEhH02dELb4-zeLcVzhoqON60cABTpbYSf1lLbYZsVUQ3cYE9CxXaByY2YNuQgc0k29mSmUvwEs0hNA5xUcE3-y_qKpYKniErb9Q"
+    }
+  ]
+}
+```
+
+Currently in local storage:
+```
+OPENID_CONFIGURATION
+RP_PRIVATE_KEY
+RP_PUBLIC_KEY
+OP_JWKS
+```
+
+#### 9. Sends Dynamic Registration Parameters
+
+Now we have everything we need to perform [dynamic client registration](https://openid.net/specs/openid-connect-registration-1_0.html). Because each Solid user could have a different OP, it is not feasible to expect the developers of RPs to manually register with every OP. Therefore, the client (RP) is registered dynamically. A request is sent to the OP's `registration_endpoint`:
+
+```
+POST https://secureauth.example/register
+```
+Data:
+```json
+{
+  grant_types: ["implicit"]
+  issuer: "https://secureauth.example"
+  redirect_uris: ["https://www.decentphotos.example/"]
+  response_types: ["id_token token"]
+  scope: "openid profile"
+}
```

Comment:
invalid JSON

Reply:
can you explain why it is invalid?  I just tried this:
```js
console.log(JSON.parse(`{\
  "grant_types": ["implicit"],\
  "issuer": "https://secureauth.example",\
  "redirect_uris": ["https://www.decentphotos.example/"],\
  "response_types": ["id_token token"],\
  "scope": "openid profile"\
}`))
```
output:
```js
{ grant_types: [ 'implicit' ],
  issuer: 'https://secureauth.example',
  redirect_uris: [ 'https://www.decentphotos.example/' ],
  response_types: [ 'id_token token' ],
  scope: 'openid profile' }
```